backends/apple/coreml/test/test_coreml_quantizer.py

```diff
@@ -0,0 +1,107 @@
+# Copyright © 2024 Apple Inc. All rights reserved.
+#
+# Please refer to the license found in the LICENSE file in the root directory of the source tree.
+
+import numpy as np
+import pytest
+from typing import Tuple
+
+import torch
+from torch._export import capture_pre_autograd_graph
+from torch.ao.quantization.quantize_pt2e import convert_pt2e, prepare_pt2e, prepare_qat_pt2e
+
+from executorch.backends.apple.coreml.quantizer.coreml_quantizer import CoreMLQuantizer
+
+from coremltools.optimize.torch.quantization.quantization_config import (
+    LinearQuantizerConfig,
+    QuantizationScheme,
+)
+
+
+class TestCoreMLQuantizer:
+    @staticmethod
+    def quantize_and_compare(
+        model,
+        example_inputs: Tuple[torch.Tensor],
+        quantization_type: str,
+    ) -> None:
+        assert quantization_type in {"PTQ", "QAT"}
+
+        pre_autograd_aten_dialect = capture_pre_autograd_graph(model, example_inputs)
+
+        quantization_config = LinearQuantizerConfig.from_dict(
+            {
+                "global_config": {
+                    "quantization_scheme": QuantizationScheme.symmetric,
+                    "milestones": [0, 0, 10, 10],
+                    "activation_dtype": torch.quint8,
+                    "weight_dtype": torch.qint8,
```

Comment:
we are deprecating qdtypes, can you use `uint8` and `int8` directly?

Reply:
Sure, updated to `uint8` and `int8`.

We will need to update coremltools for it as well, since our [quantization config](https://github.com/apple/coremltools/blob/main/coremltools/optimize/torch/quantization/quantization_config.py#L187-L211) has only qdtypes for now. Filed [coremltools issue](https://github.com/apple/coremltools/issues/2167) to track.

Reply:
[coremltools PR](https://github.com/apple/coremltools/pull/2168) merged, `uint8` and `int8` can now work with tot coremltools